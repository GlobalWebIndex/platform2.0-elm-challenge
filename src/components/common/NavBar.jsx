import React from "react";
import { Link, NavLink } from "react-router-dom";
import { Routes as RoutesConfig } from "../../constants";
import { useStyles } from "./NavBar.styles";

const NavBar = () => {
  const classes = useStyles();
  const getLinkItemClassName = (props) =>
    props.isActive ? classes.navLinkItemActive : classes.navLinkItem;

  return (
    <div className={classes.nav}>
      <div className={classes.navHeader}>
        <div className={classes.navTitle}>
          <img src="/logo.png" height="25" width="32" alt="logo" />
          <Link className={classes.navTitleText} to={RoutesConfig.home}>
            Cat lover
          </Link>
        </div>
      </div>

      <div className={classes.navLinks}>
        <NavLink className={getLinkItemClassName} to={RoutesConfig.home}>
          Home
        </NavLink>
        <NavLink className={getLinkItemClassName} to={RoutesConfig.breeds}>
          Breeds
        </NavLink>
        <NavLink className={getLinkItemClassName} to={RoutesConfig.favourites}>
          Favourites
        </NavLink>
      </div>
    </div>
  );
};

export default NavBar;
