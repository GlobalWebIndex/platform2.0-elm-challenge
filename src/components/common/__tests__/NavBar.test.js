import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import NavBar from "../NavBar";

describe("<NavBar/>", () => {
  it("renders", () => {
    const { container } = render(<NavBar />, { wrapper: BrowserRouter });

    expect(container).toMatchSnapshot();
  });
});
