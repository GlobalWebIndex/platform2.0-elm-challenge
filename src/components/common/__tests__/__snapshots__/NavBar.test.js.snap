// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar/> renders 1`] = `
<div>
  <div
    class="nav-0-2-1"
  >
    <div
      class="navHeader-0-2-2"
    >
      <div
        class="navTitle-0-2-3"
      >
        <img
          alt="logo"
          height="25"
          src="/logo.png"
          width="32"
        />
        <a
          class="navTitleText-0-2-4"
          href="/"
        >
          Cat lover
        </a>
      </div>
    </div>
    <div
      class="navLinks-0-2-5"
    >
      <a
        aria-current="page"
        class="navLinkItemActive-0-2-7 navLinkItem-0-2-6"
        href="/"
      >
        Home
      </a>
      <a
        class="navLinkItem-0-2-6"
        href="/breeds"
      >
        Breeds
      </a>
      <a
        class="navLinkItem-0-2-6"
        href="/favourites"
      >
        Favourites
      </a>
    </div>
  </div>
</div>
`;
